src/tabs/Tabs.tsx

```diff
@@ -1,34 +1,158 @@
+        // else the item remains visible and its ref from the dropdown list will be hidden otherwise
+        const hideIndexes: number[] = [];
+        let itemsWidth = 0;
+        for (let i = 0; i < items.length; i++) {
+            itemsWidth += items[i].width;
+            if (itemsWidth >= clientWidth) {
+                items[i].element.setAttribute('style', 'display: none');
+            } else {
+                hideIndexes.push(i);
+                items[i].element.removeAttribute('style');
+            }
+        }
+        setHideIndexes(hideIndexes);
+        console.log('items hidden:', items.length - hideIndexes.length);
+    }
+
+    useEffect(() => {
+        if (orientation === 'horizontal' && ref.current) {
+            // fetch the root node of tabs
+            const root = Array.from((ref.current as HTMLButtonElement).children).find(e => e.className.includes('tabRoot'));
+            if (root) {
+                // from the root node, traverse down to list node of tab items
+                const list = Array.from(root.children).find(e => e.className.includes('tabList'));
+                const items = list ? Array.from(list.children).map(element => ({ element, width: element.clientWidth })) : [];
+                setItems(items);
+
+                // apply the resize observer to the root node and watch for the size changes
+                const resizeObserver = new ResizeObserver(() => handleTabOverflow(root, items));
+                resizeObserver.observe(root);
+                return () => resizeObserver.disconnect();
+            }
+        }
+    }, [ref, orientation])
+
+
+    function handleOverflowTabChange(ev: SyntheticEvent, index: number) {
+        if (items[index]) {
+            (items[index].element as HTMLButtonElement|HTMLAnchorElement).click();
+        }
+    }
+
+    function renderMuiTabs() {
+        return (
+            <MuiTabs
+                ref={ref}
+                orientation={orientation}
+                visibleScrollbar={false}
+                scrollButtons={false}
+                allowScrollButtonsMobile={false}
+                sx={{
+                    ...sx,
+                    flex: orientation == "vertical" ? undefined : 1,
+                    borderRight: orientation == "vertical" ? 1 : 0,
+                    borderBottom: orientation == "horizontal" ? 1 : 0,
+                    borderColor: "divider"
+                }}
+                classes={{
+                    scroller: 'tabRoot',
+                    flexContainer: 'tabList'
+                }}
+                onChange={onChange}
+                {...rest}
+            />
+        );
+    }
+
+    if (orientation === 'vertical') return renderMuiTabs();
 
     return (
-        <MuiTabs
-            orientation={orientation}
-            variant={variant}
-            scrollButtons={scrollButtons}
-            allowScrollButtonsMobile={allowScrollButtonsMobile}
+        <Box
```

Comment:
This approach might work without an additional parent `Box`.

Reply:
Without the parent Box with `flex: 1`, the TabItem with a specific component (e.g.: 'a') won't be rendered correctly. Since we forced the orientation to `horizontal`, I'd like to wrap the tabs and the dropdown button within a wrapper by default in order to simplify the usage to just `<Tabs ... />`.